app/imports/api/users/server/createUser.js

```diff
@@ -40,10 +40,36 @@ const setAdminOnFirstUser = (options, user) => {
     return user;
 };
 
+const getPublicEmail = (user) => {
+    const githubEmail = [{
+        address: _.get(user, 'services.github.email', undefined),
+        verified: true,
+    }];
+    return githubEmail || [{ address: 'contact@email.com', verified: false }];
+};
+
+const getPublicUsername = (user) => {
+    const githubUsername = {
+        serviceName: _.get(user, 'services.github.username', undefined),
+        service: 'github',
+    };
+    return githubUsername || { serviceName: user.profile.name, service: 'none' };
+};
+
+const setPublicEmailAndUsername = (options, user) => {
+    if (user && user.profile && !user.emails && !user.profile.username) {
+        user.emails = getPublicEmail(user);
+        user.profile.username = getPublicUsername(user);
```

Comment:
Yes, but it's *server-side* method and here you can use private data. I think that in avatar case you can make only server-side methods, because:
1) user doesn't change his avatar every day and we can allow a little latency in this one method (see also below)
2) this implementation is very confusing... (see below)

Is your username an object? It's very confusing...
You have there at least three fields `username`, `serviceName` and `service`.... Username by definition is a string like `Avogardo` or `mstanuch`, only text. Your is an object, please tell me why, cuz I'm very confused.